cgsmiles/graph_layout.py

```diff
@@ -0,0 +1,222 @@
+"""
+Generate positions for 2D atomistic graphs.
+"""
+from collections import defaultdict, OrderedDict
+import math
+import numpy as np
+import scipy.optimize
+from scipy.spatial.distance import pdist, squareform
+import networkx as nx
+
+def find_triplets(graph):
+    """
+    Find all triplets of nodes in the graph.
+
+    Parameters
+    ----------
+    graph: nx.Graph
+
+    Returns
+    -------
+    tuple([abc.hashable,abc.hashable,abc.hashable])
+        the node keys corresponding to the triplet
+        the central node is the common one.
+    """
+    triplets = set()
```

Comment:
I kind of expected there to be a nx function to generate these, such as https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.triads.triads_by_type.html#networkx.algorithms.triads.triads_by_type, but that's only for directed graphs